libcaf_core/caf/actor_system.hpp

```diff
@@ -659,7 +660,7 @@ class CAF_CORE_EXPORT actor_system {
 
   expected<strong_actor_ptr>
   dyn_spawn_impl(const std::string& name, message& args, execution_unit* ctx,
-                 bool check_interface, optional<const mpi&> expected_ifs);
+                 bool check_interface, std::optional<const mpi*> expected_ifs);
```

Comment:
```suggestion
                 bool check_interface, const mpi* expected_ifs);
```

A pointer is already an "optional reference" since it's nullable. I don't think there's a reason to have an optional pointer.